cfpb/cfgov-refresh
# Job pages should only have a single region

Currently job listings (represented by the `jobmanager.JobListingPage` page type) can be associated with multiple geographic regions. For simplicity and to support some additional region-specific data, we want to change this so that each job is only ever associated with a single region. This PR makes the necessary changes to support that, including a migration that updates existing jobs.

## Additions

- `jobmanager.migrations.0012_jobs_have_one_region` migration converts existing jobs: if jobs have multiple regions, they are automatically assigned to the headquarters region.

## Removals

- Removed `JobListingPage` as an option in the Wagtail sidebar as it doesn't really make sense there and inserting jobs should only happen at a certain place in the tree.

## Changes

- `JobListingPage` now has a specific `region` attribute for one region only.
- Replaced the `jobmanager.Location` model with a more-appropriately named `jobmanager.JobRegion` one.

## Testing

- The best way to test this is with a production database dump:
 ```sh
 $ ./refresh-data.sh /path/to/db.sql
 $ python cfgov/manage.py migrate
 ```
- Browse through existing job pages and notice that each one has an assigned region, and that regions have been assigned properly.
- Try to create a new `JobListingPage` and notice that the region field is required.
- Preview or view existing job pages to observe how the region is displayed.

## Screenshots

The region field on a job listing now always appears like this:

![image](https://cloud.githubusercontent.com/assets/654645/21867278/7bce4ee0-d81b-11e6-92f8-bea4581771c0.png)

Existing regions can be viewed or edited through the Wagtail sidebar:

![image](https://cloud.githubusercontent.com/assets/654645/21867292/8d582a6e-d81b-11e6-8005-949744d0fc6a.png)

![image](https://cloud.githubusercontent.com/assets/654645/21867298/95845960-d81b-11e6-8549-7b7cfa90c56f.png)

## Checklist
